### PR TITLE
✨ Add async notifications

### PR DESCRIPTION
**Pull request by 5ebastianMeier**
**Changes:**

2. Added `onUserTaskWaiting` to `IManagementApiService` to subscribe to `UserTaskWaitingMessage`.
3. Added `onUserTaskFinished` to `IManagementApiService` to subscribe to `UserTaskFinishedMessage`.
4. Added `onProcessEnded` to `IManagementApiService` to subscribe to `ProcessEndedMessage`.
5. Added `onProcessTerminated` to `IManagementApiService` to subscribe to `ProcessEndedMessage`.
7. Added `socketSettings` to have a singular point for route and namespace declarations used in the `ManagementApiSocketEndpoint`.

**Issues:**

https://github.com/process-engine/process_engine_runtime/issues/71

PR: #20 

---

## How can others test the changes?

See https://github.com/process-engine/consumer_api_meta/pull/48.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

**Comment by ElRaptorus** [comment]
https://github.com/process-engine/process_engine_runtime/issues/71

File: src/imanagement_api_service.ts [hunk 1]
```diff
@@ -294,4 +298,11 @@ export interface IManagementApiService {
                                processModelId: string,
                                correlationId: string,
                                flowNodeId: string): Promise<Array<TokenHistoryEntry>>;
+
+
+
+  onUserTaskWaiting(callback: (userTaskWaiting: UserTaskWaitingMessage) => void|Promise<void>): void;
```
Comment (ElRaptorus) [inline src/imanagement_api_service.ts]: Please move `(userTaskWaiting: UserTaskWaitingMessage) => void|Promise<void>` to a type. 
Makes the method signatures far easier to read.

File: src/socket_settings.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// tslint:disable:typedef
+const paths = {
+  userTaskWaiting: `user_task_waiting`,
+  userTaskFinished: `user_task_finished`,
+  processEnded: `procses_ended`,
```
Comment (cmg-dev) [inline src/socket_settings.ts]: Correct Typo: `process_ended`

File: src/socket_settings.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// tslint:disable:typedef
+const paths = {
```
Comment (cmg-dev) [inline src/socket_settings.ts]: 👍

File: src/imanagement_api_service.ts [hunk 4]
```diff
@@ -294,4 +298,11 @@ export interface IManagementApiService {
...
+  onUserTaskFinished(callback: (userTaskFinished: UserTaskFinishedMessage) => void|Promise<void>): void;
+  onProcessTerminated(callback: (processEnded: ProcessEndedMessage) => void|Promise<void>): void;
+  onProcessEnded(callback: (processEnded: ProcessEndedMessage) => void|Promise<void>): void;
```
Comment (cmg-dev) [inline src/imanagement_api_service.ts]: The same for all other signatures.

Maybe for
`(processEnded: ProcessEndedMessage) => void|Promise<void>`

Just say `processEndedCallback`

File: src/imanagement_api_service.ts [hunk 5]
```diff
@@ -294,4 +298,11 @@ export interface IManagementApiService {
                                processModelId: string,
                                correlationId: string,
                                flowNodeId: string): Promise<Array<TokenHistoryEntry>>;
+
+
+
+  onUserTaskWaiting(callback: (userTaskWaiting: UserTaskWaitingMessage) => void|Promise<void>): void;
+  onUserTaskFinished(callback: (userTaskFinished: UserTaskFinishedMessage) => void|Promise<void>): void;
+  onProcessTerminated(callback: (processEnded: ProcessEndedMessage) => void|Promise<void>): void;
```
Comment (cmg-dev) [inline src/imanagement_api_service.ts]: Shouldn't this be `processTerminated`?

**Pull request merged**